drivers/volume/portworx/portworx.go

```diff
@@ -193,6 +193,14 @@ func (d *portworx) InspectVolume(name string, params map[string]string) error {
 		}
 	}
 
+	// Labels
+	if err := d.schedOps.ValidateLabels(vol); err != nil {
```

Comment:
Let's have a generic API `ValidateVolume` and as part of the implementation we will validate the labels.

Reply:
We should actually have two APIs
- ValidateCreateVolume - checks if labels are applied on the node
- ValidateDeleteVolume - checks if labels are removed from the node.

Reply:
@harsh-px why not rename the InspectVolume to ValidateCreateVolume as aditya suggested, as it is doing the same. And keep the ValidateAddLabels (ValidateLabels) as part of it?
@adityadani I will add a ValidateDeleteVolume which will currently check the label removal, but as you suggested we can also verify that the volume is not present in px volume list call, etc.. probably will do that as part of a later check-in

Reply:
Yes, add `InspectVolume` as a method in the schedops interface.

For volume delete validation, it can be merged with the existing `ValidateVolumeCleanup` method in the schedops interface. You can rename `ValidateVolumeCleanup` to `ValidateDeleteVolume`.

Reply:
The issue here is ValidateVolumeCleanup is called after suite and does not check volume specific checks. We will have to add a separate ValildateDeleteVolume in portworx.go which checks if a given volume is deleted/cleanup or not.

Reply:
Sure. Go ahead with `ValildateDeleteVolume`.